lib/PuppeteerSharp/Page.cs

```diff
@@ -667,25 +667,29 @@ public async Task<byte[]> ScreenshotDataAsync(ScreenshotOptions options)
         public Task<string> GetTitleAsync() => MainFrame.GetTitleAsync();
 
         /// <inheritdoc/>
-        public Task CloseAsync(PageCloseOptions options = null)
+        public async Task CloseAsync(PageCloseOptions options = null)
         {
-            if (!(Client?.Connection?.IsClosed ?? true))
+            if (Client?.Connection?.IsClosed ?? true)
             {
-                var runBeforeUnload = options?.RunBeforeUnload ?? false;
+                _logger.LogWarning("Protocol error: Connection closed. Most likely the page has been closed.");
+                return;
+            }
 
-                if (runBeforeUnload)
-                {
-                    return Client.SendAsync("Page.close");
-                }
+            var runBeforeUnload = options?.RunBeforeUnload ?? false;
 
-                return Client.Connection.SendAsync("Target.closeTarget", new TargetCloseTargetRequest
+            if (runBeforeUnload)
+            {
+                await Client.SendAsync("Page.close").ConfigureAwait(false);
+            }
+            else
+            {
+                await Client.Connection.SendAsync("Target.closeTarget", new TargetCloseTargetRequest
                 {
                     TargetId = Target.TargetId,
-                }).ContinueWith(task => ((Target)Target).CloseTask, TaskScheduler.Default);
-            }
+                }).ConfigureAwait(false);
 
-            _logger.LogWarning("Protocol error: Connection closed. Most likely the page has been closed.");
-            return Task.CompletedTask;
+                await ((Target)Target).CloseTask.ConfigureAwait(false);
```

Comment:
I'm noticing that we now only execute `((Target)Target).CloseTask` if `Client.Connection.SendAsync("Target.closeTarget"` completes _successfully_.
Should we still execute `((Target)Target).CloseTask` if `Client.Connection.SendAsync("Target.closeTarget"` throws an exception?

Reply:
@jnyrup this matches [upstream](https://github.com/puppeteer/puppeteer/blob/main/packages/puppeteer-core/src/common/Page.ts#L1506)